src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,31 +36,47 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        // Split into different parts of address
+        String[] addressParts = trimmedAddress.split(SEPARATOR);
+        block  = new Block(addressParts[PART_BLOCK].trim());
+        street = new Street(addressParts[PART_STREET].trim());
+        unit   = new Unit(addressParts[PART_UNIT].trim());
+        postalCode = new PostalCode(addressParts[PART_POSTALCODE].trim());
     }
 
     /**
      * Returns true if a given string is a valid person address.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    private static boolean isValidAddress(String test) {
```

Comment:
This is a very weak validation; It can take any gibberish as long as it has 4 parts!

Reply:
Will take note and use proper Regex next time